### ISSUE TITLE
postfix seems to not known its own domain

### ISSUE THREAD

**Issue by nsk7even** [tomav/docker-mailserver#1522]
## Subject

* [x] Other

## Problem in short:

Postfix tries to relay all mail

## Maybe the culprit:

- My system runs on a subdomain: mail.mydomain.de
- This is my homeserver and I have set up the domain server of mydomain.de to point mail.mydomain.de to a freedns.afraid.org dynamic subdomain name

### Relay disabled:

- Local delivery OK
- IN delivery OK
- OUT delivery NOT working (because of no permission to create SPF records for a freedns.afraid.org subdomain name and a TXT record for mail.mydomain.de seems to not be respected)

### Relay enabled:

- Local delivery NOT working (see below)
- IN delivery NOT working (see below)
- OUT delivery OK

### Local delivery and IN delivery, when relay is enabled:

```
postfix/smtp[5508]: warning: smtp_tls_wrappermode requires "smtp_tls_security_level = encrypt" (or stronger)
May 21 22:56:46 enceladus postfix/smtp[3233]: 4F25B80B76: to=<me@mail.mydomain.de>, relay=none, delay=513, delays=513/0.02/0/0, dsn=4.3.0, status=deferred (server unavailable or unable to receive mail)
```

### .env

```
HOSTNAME=enceladus
DOMAINNAME=mail.mydomain.de
CONTAINER_NAME=mail
```

### env-mailserver

```
OVERRIDE_HOSTNAME=enceladus.mail.mydomain.de
RELAY_HOST=relayhost
RELAY_PORT=465
RELAY_USER=myuser
RELAY_PASSWORD=mypwd
```

### config/postfix-main.cf

`smtp_tls_wrappermode = yes`

### postfix-relaymap.cf

```
@mail.mydomain.de	[relayhost]:465
*			DUNNO
```

## Question

So the question is, did I confuse postfix with my setup/environment somehow in a way that it doesn't know its local domain?
Or did I got this wrong and the relay setup just is not correct?

Thanks for any help! :)

**Comment by erik-wramner** [comment]
Well, I'm no expert on configurations with relay, I've never used that part. However, your domain seems weird. Do you send mail to `someuser@mail.mydomain.com` or do you actually send mail to `someuser@mydomain.com`?

My guess is that the domain should be `mydomain.com` but of course your mail server (and the MX record) can use `somehost.mail.mydomain.com` if it wants. If you can't control the DNS enough to use `mydomain.com` I think you have a problem as most users will send mail to that address?

**Comment by nsk7even** [comment]
Yes, mail is sent to `someuser@mail.mydomain.com`. `mydomain.com` has its own mail server (from my hosting provider) and I use this one primarily, including mail addresses like `userx@mydomain.com`.

But I want to establish my home server as mail server in parallel, using the subdomain. I can make DNS entries for `mydomain.com` and `mail.mydomain.com` with no problems.
DKIM e. g. worked without a problem there, cause the name `mail._domainkey.mail.mydomain.com` is queried from the nameservers of mydomain.com.

But name translation of records from `mail.mydomain.com` seems to query the freedns.afraid.org DNS servers, as they follow the CNAME entry that points `mail.mydomain.com` to `me.somedynamic.domain`.

There seems to be no problem, as everything is working, except for the relay mechanism.
The fact that the relay, when enabled, works as well and it is just to greedy, makes me think that it should be possible to getting this working somehow.... :)

**Comment by erik-wramner** [comment]
It should be doable, I'll flag it help wanted. Hopefully someone with more experience in relay configuration will notice.

**Comment by aendeavor** [comment]
There hasn't happened much here. Any news @nsk7even ?

**Comment by nsk7even** [comment]
No, nothing yet, but I will try this again, as soon as I have some spare times! (hopefully not later than in winter)

**Comment by aendeavor** [comment]
@nsk7even please report back here when you're done:)

**Comment by aendeavor** [comment]
What's the current status here?

**Comment by nsk7even** [comment]
> What's the current status here?

still no progress here, was quite busy over the last months

**Comment by aendeavor** [comment]
Alright. We're currently cleaning the issue tracker due to migration. Therefore, I'll close this issue, but I wish you luck with your use case.
